### PR TITLE
Add sample for cron

### PR DESCRIPTION
**Pull request by vancexu**

### CODE REVIEW

File: src/main/java/com/uber/cadence/samples/hello/HelloCron.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    @WorkflowMethod(
+      // At most one instance.
+      workflowId = CRON_WORKFLOW_ID,
+      taskList = TASK_LIST,
+      // timeout for every run
+      executionStartToCloseTimeoutSeconds = 30,
+      // To allow starting workflow with the same ID after the previous one has terminated.
+      workflowIdReusePolicy = WorkflowIdReusePolicy.AllowDuplicate
+    )
+    @CronSchedule("*/1 * * * *") // new workflow run every minute
```
Comment (mkolodezny) [inline src/main/java/com/uber/cadence/samples/hello/HelloCron.java:55]: nit: is 'every min' the least interval we can support for java/go now?

File: src/test/java/com/uber/cadence/samples/hello/HelloCronTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+  @After
+  public void tearDown() {
+    testEnv.close();
+  }
+
+  @Test
+  public void testCron() throws TException {
+    worker.registerActivitiesImplementations(new HelloCron.GreetingActivitiesImpl());
+    testEnv.start();
```
Comment (mkolodezny) [inline src/test/java/com/uber/cadence/samples/hello/HelloCronTest.java:62]: nit: does testEnv support a time machine so no need to wait minutes here?

File: src/main/java/com/uber/cadence/samples/hello/HelloCron.java (hunk 1 above)
Comment (vancexu) [reply src/main/java/com/uber/cadence/samples/hello/HelloCron.java:55]: No we only support standard cron syntax. Minute is the least interval. 
For seconds cron, use timer and continueAsNew.

File: src/test/java/com/uber/cadence/samples/hello/HelloCronTest.java (hunk 2 above)
Comment (vancexu) [reply src/test/java/com/uber/cadence/samples/hello/HelloCronTest.java:62]: no unit test use logic clock to avoid waiting.

**Pull request merged**